internal/condition/condition.go

```diff
@@ -181,7 +183,7 @@ func (e *EvaluableCondition) CastContextToTypedParameters(contextMap map[string]
 // context provided. If more than one source of context is provided, and if the
 // keys provided in those context(s) are overlapping, then the overlapping key
 // for the last most context wins.
-func (e *EvaluableCondition) Evaluate(contextMaps ...map[string]*structpb.Value) (EvaluationResult, error) {
+func (e *EvaluableCondition) Evaluate(ctx context.Context, contextMaps ...map[string]*structpb.Value) (EvaluationResult, error) {
```

Comment:
You can also add

```
if ctx.Err() != nil {
   return nil, ctx.Err()
}
```

Reply:
@miparnisari why do you think we should do this here?

Reply:
In this case though I don’t see the value. We’re not potentially invoking I/O heavy workloads between the top of the function body and where the context cancellation would actually cancel the costly operation `ContextEval` . I wouldn’t encourage it in this scenario.

This opinion differs from the usage of this similar pattern in `internal/graph/check.go` because in that usage we're potentially avoiding several I/O heavy expansion workloads by doing the point in time context cancellation check.

Reply:
> @miparnisari why do you think we should do this here?

in case the evaluation takes up a lot of CPU or memory, we can prevent it from happening

Reply:
That's why we're adding `e.celProgram.ContextEval` though. The evaluation of the program is the costly bit, and that's what needs to be interrupted.

Reply:
Oh lol, never mind